cli/flox-rust-sdk/src/models/environment/core_environment.rs

```diff
@@ -884,6 +898,111 @@ impl CoreEnvironment<ReadOnly> {
         self.replace_with(temp_env)?;
         Ok(store_path)
     }
+
+    /// Should not be called with
+    /// !migration_info.needs_manifest_migration && !migration_info.needs_upgrade
+    pub fn migrate_to_v1(
+        &mut self,
+        flox: &Flox,
+        mut migration_info: MigrationInfo,
+    ) -> Result<PathBuf, CoreEnvironmentError> {
+        let tempdir = tempfile::tempdir_in(&flox.temp_dir)
+            .map_err(CoreEnvironmentError::MakeSandbox)?
+            .into_path();
+
+        debug!(
+            "migration transaction: making temporary environment in {}",
+            tempdir.display()
+        );
+        let mut temp_env = self.writable(&tempdir)?;
+
+        if migration_info.needs_manifest_migration {
+            migration_info
+                .raw_manifest
+                .insert(MANIFEST_VERSION_KEY, toml_edit::value(1));
+
+            // Make sure it parses
+            migration_info
+                .raw_manifest
+                .to_typed()
+                .map_err(CoreEnvironmentError::MigrateManifest)?;
+
+            debug!("migration transaction: updating manifest");
+            temp_env.update_manifest(migration_info.raw_manifest.to_string())?;
+        }
+
+        // Lock if there's a v0 manifest, regardless of whether there's a
+        // lockfile or what version it is.
+        // This could lock an environment that isn't already locked,
+        // but we don't want to perform a transaction without locking.
```

Comment:
Why don't we want to perform this transaction without locking?

Reply:
Changing to
```
        // This could lock an environment that isn't already locked,
        // but particularly for managed and remote environments, we want to keep
        // the lock in sync with the manifest,
        // so we don't want to perform a transaction without locking.
```